Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/DefaultChartDataCollection.java

```diff
@@ -189,125 +189,139 @@ public DefaultChartDataCollection(APMLog apmLog) {
          */
         TreeSortedMap<Double, Integer> ttlPTMap = new TreeSortedMap<>();
         double ttlPTUnit = 0;
-        if(interval > 0) {
-            Collections.sort(allTotalProcessingTimes);
-            int numOfUnits = MAX_UNIT;
-            double first = allTotalProcessingTimes.get(0);
-            double last = allTotalProcessingTimes.get(allTotalProcessingTimes.size() - 1);
-            ttlPTUnit = (last - first) / numOfUnits;
-            double current = first - ttlPTUnit;
-            for(int i=0; i <= numOfUnits; i++) {
-                if(i == numOfUnits) current = last;
-                else current += ttlPTUnit;
-                ttlPTMap.put(current, 0);
+        if (allTotalProcessingTimes.size() > 0) {
+            if (interval > 0) {
+                Collections.sort(allTotalProcessingTimes);
+                int numOfUnits = MAX_UNIT;
+                double first = allTotalProcessingTimes.get(0);
+                double last = allTotalProcessingTimes.get(allTotalProcessingTimes.size() - 1);
+                ttlPTUnit = (last - first) / numOfUnits;
+                double current = first - ttlPTUnit;
+                for (int i = 0; i <= numOfUnits; i++) {
+                    if (i == numOfUnits) current = last;
+                    else current += ttlPTUnit;
+                    ttlPTMap.put(current, 0);
+                }
             }
         }
         /**
          * For average processing time
          */
         TreeSortedMap<Double, Integer> avgPTMap = new TreeSortedMap<>();
         double avgPTUnit = 0;
-        if(interval > 0) {
-            Collections.sort(allAverageProcessingTimes);
-            double first = allAverageProcessingTimes.get(0);
-            double last = allAverageProcessingTimes.get(allAverageProcessingTimes.size() - 1);
-            avgPTUnit = (last - first) / MAX_UNIT;
-            double current = first - avgPTUnit;
-            for(int i=0; i <= MAX_UNIT; i++) {
-                if(i == MAX_UNIT) current = last;
-                else current += avgPTUnit;
-                avgPTMap.put(current, 0);
+        if (allAverageProcessingTimes.size() > 0) {
+            if (interval > 0) {
+                Collections.sort(allAverageProcessingTimes);
+                double first = allAverageProcessingTimes.get(0);
```

Comment:
Duplicate Code. Can use. a template method, which does all the above process and then just.
1 liner to do average

Reply:
Will do this in separate task